lib/routes/utils/request_helper.js

```diff
@@ -9,9 +9,32 @@
  * @returns {boolean}
  */
 function wantsKeys (request) {
-  return request.query.keys === 'true'
+  return request.query.keys === 'true' || request.query.keys === true
+}
+
+/**
+ * Returns whether or not to perform a signin verification email.
+ * Currently, the uid is converted to a number between 0-100 which corresponds to the
+ * sign-in confirmation sample rate.
+ *
+ * @param uid
+ * @returns {boolean}
+ */
+function shouldEnableSigninConfirmation(uid, config) {
+  // Sign-in confirmation is determined by whether the feature is enabled and
+  // if the user's first two uid digits fall within the config sample rate.
+  if (config.signinConfirmation && config.signinConfirmation.enabled) {
+
+    // Returns the first 2 integers of uid
+    var uidNum = parseInt(uid.substr(0, 4), 16) % 100
```

Comment:
This seems to rely on the most significant digits. Do we know whether, in practice, that will return an even spread of values to base the sample rate on?

By definition, as numbers get bigger, the most significant digits will vary less and less. You get far more entropy from the least significant digits. Since we're already using a modulo operation here, that can be achieve by just using `parseInt(uid) % 100` and ditching the `substr`.

Does that make sense or am I worrying about nothing here?